docs/e2e.md

```diff
@@ -0,0 +1,234 @@
+    handles a variety of workload and deployer configurations.
+
+The Exhaustive Test Suite uses three primary concepts:
+
+- **workload**
+    - Deployment, DaemonSet, StatefulSet
+    - Currently, only **Deployment** is supported.
+- **deployer**
+    - Deployer refers to how to deploy the application into the cluster.
+    Available deployers:
+        - **Subscription**
+        - **ApplicationSet**
+        - **DiscoveredApp**
+- **actions**
+    - Refers to actions related to workloads, currently supported actions are:
+        - **deploy**: deploy workload in the cluster
+        - **enable**: enable workload to be protected by RamenDR
+        - **failover**: failover workload from one cluster to the other
+        - **relocate**: relocate workload from one cluster to the other
+        - **disable**: remove workload from protection by RamenDR
+        - **undeploy**: remove workload from the cluster
+
+## Overview of `config.yaml` File
+
+By default, the `config.yaml.sample` contains following configurations:
+
+`$ cat e2e/config.yaml.sample`
+
+```yaml
+channelname: "ramen-gitops"
+channelnamespace: "ramen-samples"
+giturl: "https://github.com/RamenDR/ocm-ramen-samples.git"
+pvcspecs:
+  - storageclassname: rook-cephfs
+    accessmodes: ReadWriteMany
+  - storageclassname: rook-ceph-block
+    accessmodes: ReadWriteOnce
+```
+
+### Explanation of Config Fields
+
+- `channelname`:
+    Name of the channel for the channel CR. The channel CR
+    is later used by subscription CR.
+
+- `channelnamespace`:
+    Namespace in which the channel CR is created.
+
+- `giturl`:
+    Git URL pointing to the repository with application manifests.
+
+    Deployment uses below fixed values, and git url defined above,
+    pointing to an application in the git url to be deployed on the cluster:
+
+    ```
+    GITPATH     = "workloads/deployment/base"
+    GITREVISION = "main"
+    APPNAME     = "busybox"
+    ```
+
+- `pvcspecs`:
+    are used to kustomize the workload to use different pvc
+    parameters like `storageclassname` and `accessmodes`.
+
+    Each pvc specification in the list of **pvcspecs** will be applied
+    for each workload. By default, it will creates two busybox deployments
+    with different pvc parameters, one with CephFS and RWX and the other
+    with RBD and RWO.
+
+## Prepare `config.yaml` File
+
+- Create a config.yaml file by copying `config.yaml.sample` template:
+
+    `$ cat e2e/config.yaml.sample >> e2e/config.yaml`
+
+- Prepare config.yaml by adding kubeconfig file path, with format:
+
+    ```yaml
+    Clusters:
+        c1:
+            kubeconfigpath: /path/to/kubeconfig/c1
+        c2:
+            kubeconfigpath: /path/to/kubeconfig/c2
+        hub:
+            kubeconfigpath: /path/to/kubeconfig/hub
+    ```
+
+    If `drenv` was used to configure RDR clusters, easily add the clusters
+    kubeconfig file paths into `config.yaml` by:
+
+    `$ cat ~/.config/drenv/rdr/config.yaml >> config.yaml`
```

Comment:
This style, having intended code under list items is hard to maintain and does not support all markup features (like alerts), so we want to avoid it.

We can use either new heading for each step:

    ## Prepare config file

    ### Copy the file

    ```console
    command...
    ```
    
    ### Add clusters

    ```yaml
    Clusters:
      ...
    ```

Or use commands example with text describing them:

    ## Prepare config file

    Copy the file

    ```
    command...
    ```   

    Add clusters:

    ```yaml
    Clusters:
      ...
    ```

Using new heading is better when we have more content, if we describe few simple commands it they do not add much value.

Reply:
ack, updated.